docs/adr/0038-use-component-library.md

```diff
@@ -0,0 +1,59 @@
+# Use a React component library
+
+- Status: proposed
+- Deciders: Reino Muhl
+- Date: 2023-04-03
+
+## Context and Problem Statement
+
+As part of a major refactor of the Subscription Platform, the team is trying to simplify and reduce the number of components used by the Subscription Platform web app. To that end the team is investigating whether or not to replace some of the more commonly used components, such as a Tooltip, DialogModal, etc., with a component library that provides out of the box features such as accessibility and screen reader support, for example.
+
+Additionally, maintaining and implementing the existing components sometimes comes with additional developer cost. For example, during a recent refactor introducing TailwindCSS, it was difficult to refactor items in the bespoke component library due to deeply nested styling.
+
+## Decision Drivers
+
+- Accessibility
+- Speed of development
+- Maintainability
+
+## Considered Options
+
+- A. Unstyled component library
+- B. Styled component library
+- C. Bespoke components
+
+## Decision Outcome
+
+Chosen option: A. Unstyled component library. Using an unstyled component library, also sometimes referred to as a headless component library, meets all key decision drivers, with few draw backs. Unstyled components allows the team to bring their own existing styling library, TailwindCSS, and brings
```

Comment:
Re: "Chosen option: A. Unstyled component library"

Do we have a chosen component library already, or is that a separate, future ADR?

Reply:
Yup, I was thinking maybe a future ADR or maybe team agreement is good enough.

I thought it might be best to first discuss the general approach before discussing specific libraries.

Reply:
An unstyled component library sounds really enticing! It might be helpful to know if the front runners offer components that align well with those used/needed by fxa (e.g., would the libraries cover our needs pretty well, or would we still need to create a lot of bespoke components?).

Reply:
> and provide accessibility features etc. out of the box

I would have questions around this per library/possibly per component. Do the frontrunner libraries (Headless UI it looks like) allow us to customize a11y options/add any props or make adjustments if we disagree with something or just see improvements to be made? I'm guessing abstracting a11y away would be worth it, but, I'm wondering if it sacrifices flexibility too since some a11y is subjective and can depend on other content on the page. Maybe you're planning on having @sardesam evaluate and determine if a library/components are good on this front.

It looks like Headless UI only offers [10 components](https://headlessui.com/) right now which is interesting. Which ones do we foresee SubPlat using? I know this ADR covers if we should use a library or not and not necessary the library itself, but I feel like they kind of go hand-in-hand. I could agree with the points in this ADR but then look at library options later and be like "eh never mind, none of them hit all the points we were looking to solve in the first ADR".